fido/Crypto/Fido2/Assertion.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Crypto.Fido2.Assertion
+  ( Error (..),
+    verifyAssertionResponse,
+    RelyingPartyConfig (..),
+    Credential (..),
+  )
+where
+
+import Control.Monad (when)
+import qualified Crypto.Fido2.Protocol as Fido2
+import qualified Crypto.Hash as Hash
+import qualified Data.ByteArray as BA
+import qualified Data.List as List
+import qualified Data.Text.Encoding as Text
+
+data Error
+  = CredentialDoesNotMatch
+  | UnsupportedClientDataType
+  | ChallengeMismatch
+  | OriginMismatch
+  | RpIdMismatch
+  | UserNotPresent
+  | UserNotVerified
+  | InvalidSignature
+  | -- TODO: Get rid of this constructor as it signals a bug in our lib.
+    RawDataUnavailable
+  deriving (Show, Eq)
+
+-- Domain type for the users to use, because the types from Protocol
+-- are killing us.
+data Credential = Credential {id :: Fido2.CredentialId, publicKey :: Fido2.PublicKey}
+
+-- Domain type: configuration for our relying party.
+data RelyingPartyConfig = RelyingPartyConfig {origin :: Fido2.Origin, rpId :: Fido2.RpId}
+
+-- | Verify that a @Fido2.PublicKeyCredential@ is valid for the given
+-- @RelyingPartyConfig@, @Challenge@, and list of @Creedential@s known to be
```

Comment:
If you use ` 'RelyingPartyConfig' ` instead of `@ @ ` you get hyperlinks in the source code.

However; we should probably just do a big haddock pass when this library is ready for release, so dont bikeshed too much about it now

Reply:
Thanks! Cleaned up some minor things about the haddocks.